src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1134,56 +1188,117 @@ private void RunLayoutGraph(GraphLayout.Graph graph)
             graph.SetGraphPosition();
         }
 
-        private void SaveLayoutGraph(GraphLayout.Graph graph)
+        private void AvoidSubgraphOverlap()
+        {
+            var VerticalGraphDistance = 30;
+            var HorizontalGraphDistance = 70;
+
+            bool done;
+
+            do
+            {
+                done = true;
+
+                foreach (var g1 in Model.LayoutSubgraphs.Skip(1))
+                {
+                    foreach (var g2 in Model.LayoutSubgraphs.Skip(1))
+                    {
+                        if (!g1.Equals(g2) && (g1.GraphCenterY + g1.OffsetY <= g2.GraphCenterY + g2.OffsetY))
+                        {
+                            var g1nodes = g1.Nodes.OrderBy(n => n.Y + n.Height);
+                            var g2nodes = g2.Nodes.OrderBy(n => n.Y);
+
+                            foreach (var node1 in g1nodes)
+                            {
+                                foreach (var node2 in g2nodes)
+                                {
+                                    if ((node1.Y + node1.Height + VerticalGraphDistance + g1.OffsetY > node2.Y + g2.OffsetY) &&
+                                        (((node1.X <= node2.X) && (node1.X + node1.Width + HorizontalGraphDistance > node2.X)) ||
+                                        ((node2.X <= node1.X) && (node2.X + node2.Width + HorizontalGraphDistance > node1.X))))
+                                    {
+                                        g1.OffsetY -= 5;
+                                        g2.OffsetY += 5;
+                                        done = false;
+                                    }
+                                    if (!done) break;
+                                }
+                                if (!done) break;
+                            }
+                        }
+                    }
+                }
+            } while (!done);
+        }
+
+        private void SaveLayoutGraph()
         {
             // Assign coordinates to nodes inside groups
             foreach (var group in Model.Annotations)
             {
-                var g = graph.FindNode(group.GUID);
-                double deltaX = g.X - group.X;
-                double deltaY = g.Y - group.Y;
-                foreach (var node in group.SelectedModels.OfType<NodeModel>())
+                GraphLayout.Node g = null;
```

Comment:
I think the name `g` for node is confusing with graph. Can we change it to `n` or something?